File: app/src/main/java/kr/co/company/capstone/activity/MainActivity.java
package kr.co.company.capstone.activity;

// Android 라이브러리 import
import android.content.DialogInterface;
import android.content.Intent;
import android.os.Bundle;
import android.util.Log;

// androidx 라이브러리 import
import androidx.appcompat.app.AlertDialog;
import androidx.appcompat.app.AppCompatActivity;
import androidx.navigation.NavController;
import androidx.navigation.Navigation;
import androidx.navigation.ui.AppBarConfiguration;
import androidx.navigation.ui.NavigationUI;

// Google 라이브러리 import
import com.google.android.material.bottomnavigation.BottomNavigationView;

// Java 라이브러리 import
import java.util.List;

// kr.co.company.capstone 패키지 내 클래스 import
import kr.co.company.capstone.R;
import kr.co.company.capstone.dto.ErrorMessage;
import kr.co.company.capstone.dto.NotificationDetailListResponse;
import kr.co.company.capstone.dto.NotificationDetailResponse;
import kr.co.company.capstone.service.NotificationService;

// org.jetbrains.annotations 라이브러리 import
import org.jetbrains.annotations.NotNull;

// retrofit2 라이브러리 import
import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class MainActivity extends AppCompatActivity {

    private static final String LOG_TAG = "MainActivity";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_bottom);

        setUpNavigation();

        // 애러 다이얼로그 생성
        AlertDialog alertDialog = createOnErrorDialog();

        // 사용자 목표 수행 완료 알림이 있다면 받아옴
        NotificationService.getService().findGoalCompleteNotifications()
                .enqueue(new Callback<NotificationDetailListResponse>() {
                    @Override
                    public void onResponse(Call<NotificationDetailListResponse> call, Response<NotificationDetailListResponse> response) {
                        notificationExistOrNot(response);
                    }
                    @Override
                    public void onFailure(Call<NotificationDetailListResponse> call, Throwable t) {
                        showOnErrorDialog(alertDialog);
                    }
                });
    }

    // AlertDialog 생성, 설정
    @NotNull
    private AlertDialog createOnErrorDialog() {
        AlertDialog.Builder builder = new AlertDialog.Builder(this);
        builder.setTitle("띵").setMessage("오류가 발생했습니다.").setPositiveButton("넵", new DialogInterface.OnClickListener() {
            @Override
            public void onClick(DialogInterface dialog, int which) {
                Intent intent = getIntent();
                finish();
                startActivity(intent);
            }
        });
        return builder.create();
    }

    // 생성한 AlertDialog를 보여줌
    private void showOnErrorDialog(AlertDialog alertDialog){
        alertDialog.show();
    }

    private void notificationExistOrNot(Response<NotificationDetailListResponse> response) {
        if (response.isSuccessful()) {
            List<NotificationDetailResponse> notifications = response.body().getNotifications();

            // 받아야 할 목표 수행 완료 알림이 있다면
            if (notifications.size() > 0) {
                Intent intent = new Intent(getApplicationContext(), GoalCompleteNotificationActivity.class);
                intent.putExtra("info", notifications.get(0));
                startActivity(intent);
            }
        } else {
            Log.d(LOG_TAG, "in notification , error code : " + response.code());
            Log.d(LOG_TAG, ErrorMessage.getErrorByResponse(response).toString());
        }
    }

    private void setUpNavigation() {
        BottomNavigationView navView = findViewById(R.id.bottomNav_view);

        // AppBarConfiguration 설정
        AppBarConfiguration appBarConfiguration = new AppBarConfiguration.Builder(R.id.navigation_home, R.id.navigation_history, R.id.navigation_alarm, R.id.settingPageFragment).build();

        // NavController 초기화
        NavController navController = Navigation.findNavController(this, R.id.navHostFragment);

        // NavController와 BottomNavigationView 연결
        NavigationUI.setupWithNavController(navView, navController);


    }

}

File: app/src/main/java/kr/co/company/capstone/main/MainActivity.kt
package kr.co.company.capstone.main

import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.navigation.Navigation.findNavController
import androidx.navigation.ui.NavigationUI.setupWithNavController
import kr.co.company.capstone.R
import kr.co.company.capstone.databinding.ActivityBottomBinding

class MainActivity : AppCompatActivity(), MainContract.View {
    private var _binding : ActivityBottomBinding? = null
    private val binding get() = _binding!!
    private var presenter : MainPresenter? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        _binding = ActivityBottomBinding.inflate(layoutInflater)

        setContentView(binding.root)

        // Set Up Navigation
        setUpNavigation()

        presenter = MainPresenter(this)

        // 알림 및 외부 Intent 관리
        handleIntent(intent)
    }

    override fun setUpNavigation() {
        val navView = binding.bottomNavView

        // NavController 초기화
        val navController = findNavController(this, R.id.navHostFragment)

        // NavController와 BottomNavigationView 연결
        setupWithNavController(navView, navController)
    }

    override fun onNewIntent(intent: Intent?) {
        super.onNewIntent(intent)
        handleIntent(intent)
    }

    // 알림에 따른 동작 수행
    private fun handleIntent(intent: Intent?) {
        intent ?: return

        val navigateTo = intent.getStringExtra("navigateTo")

        val bundle = when (navigateTo) {
            // 목표 초대 메시지 처리
            "INVITE_SEND" -> Bundle().apply {
                putString("messageBody", intent.getStringExtra("messageBody"))
                putLong("notificationId", intent.getLongExtra("notificationId", 0L))
            }
            // 게시물 업로드 알림
            "POST_UPLOAD" -> Bundle().apply {
                putLong("goalId", intent.getLongExtra("goalId", 0L))
                putLong("userId", intent.getLongExtra("userId", 0L))
            }
            // 이외는 처리할 것 없음.
            else -> null
        }


        bundle?.let {
            val navController = findNavController(this, R.id.navHostFragment)
            val destinationId = when (navigateTo) {
                "INVITE_SEND" -> R.id.invitationReplyDialogFragment
                "POST_UPLOAD" -> R.id.timeLineFragment
                else -> null
            }
            destinationId?.let { navController.navigate(it, bundle) }
        }

    }
    override fun onDestroy() {
        super.onDestroy()
        _binding = null
    }
}